src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utilities for testing COS integration with charms."""
+import logging
+from pathlib import Path
+from typing import Any, Dict, Set
+
+import yaml
+from juju.action import Action
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+
+log = logging.getLogger(__name__)
+
+GRAFANA_AGENT_APP = "grafana-agent-k8s"
+METRICS_ENDPOINT_RELATION = "metrics-endpoint"
+GRAFANA_DASHBOARD_RELATION = "grafana-dashboard"
+ALER_RULES_DIRECTORY = Path("./src/prometheus_alert_rules")
+
+
+async def deploy_and_assert_grafana_agent(
+    model: Model,
+    app: str,
+    channel: str = "latest/stable",
+    dashboard: bool = True,
```

Comment:
I think that we can even set all to `False` as default, but I do not have strong opinion here.
My whole point of this was to prepare for logging (Loki), so we can use the same function. And since Loki will come first, we will use it like this:
```python
await deploy_and_assert_grafana_agent(ops_test.model, my_charm_name, logging=True, metrics=False, dashbord=False)
```
So in first iteration there will be only logging endpoint, but after the whole observability implementation I think we will have `metrics=True` and `logging=True` always and `dashboard=False` sometimes. So maybe you are right, we can do `metrics` default value True and `dashboard` False and later add `logging` as True. WDYT?

Reply:
Ok, so let's do default False for all. :+1: